src/matrixroomhandler.ts

```diff
@@ -386,6 +386,139 @@ export class MatrixRoomHandler {
+        command = allArgs[0];
+        allArgs.splice(0, 1);
+        args = allArgs;
+      }
+    }
+    
+    let replyMessage = "Error, unkown command. Try `!matrix help` to see all commands";
+    
+    const intent = this.bridge.getIntent();
+    const doAction = async (funcKey, action) => {
+      const name = args.join(" ");
+      const channels = await this.discord.GetRoomIdsFromChannel(msg.channel);
+      try {
+        const userMxid = await this.GetMxidFromName(name, channels);
+        let allChannels = [];
+        /* tslint:disable:await-promise */
+        await Bluebird.all((<Discord.TextChannel> msg.channel).guild.channels.map((c) => {
+        /* tslint:enable:await-promise */
+          return this.discord.GetRoomIdsFromChannel(c).then((chans) => {
+            allChannels = allChannels.concat(chans);
+          }).catch((e) => {
+            // pass, non-text-channel
+          });
+        }));
+        let errorMsg = "";
+        /* tslint:disable:await-promise */
+        await Bluebird.all(allChannels.map((c) => {
+        /* tslint:enable:await-promise */
+          return intent[funcKey](c, userMxid).catch((e) => {
+            // maybe we don't have permission to kick/ban/unban...?
+            errorMsg += `\nCouldn't ${funcKey} ${userMxid} from ${c}`;
+          });
+        }));
+        if (errorMsg) {
+          throw Error(errorMsg);
+        }
+        replyMessage = `${action} ${userMxid}`;
+      } catch (e) {
+        replyMessage = "**Error:** " + e.message;
+      }
+    };
+    
+    switch (command) {
+      case "help":
+        replyMessage = "Available Messages:\n" +
+            " - `!matrix kick <name>`: Kicks a user on the matrix side\n" +
+            " - `!matrix ban <name>`: Bans a user on the matrix side\n" +
+            " - `!matrix unban <name>`: Unbans a user on the matrix side\n\n" +
+            "The name must be the display name or the mxid of the user to perform the action on.";
+        break;
+      case "kick":
+        await doAction("kick", "Kicked");
+        break;
+      case "ban":
+        await doAction("ban", "Banned");
+        break;
+      case "unban":
+        await doAction("unban", "Unbanned");
+        break;
+      default:
+        break;
+    }
+    
+    msg.channel.send(replyMessage);
+  }
+
+  private async GetMxidFromName(name: string, channels: string[]) {
+    if (name[0] === "@" && name.includes(":")) {
+        return name;
+    }
+    const client = this.bridge.getIntent().getClient();
+    const matrixUsers = {};
+    let matches = 0;
+    /* tslint:disable:await-promise */
+    await Bluebird.all(channels.map((c) => {
+    /* tslint:enable:await-promise */
+      // we would use this.bridge.getBot().getJoinedMembers()
+      // but we also want to be able to search through banned members
+      // so we gotta roll our own thing
+      return client._http.authedRequestWithPrefix(
+        undefined, "GET", "/rooms/" + encodeURIComponent(c) + "/members",
```

Comment:
All these undefined's make me sad, but I don't know what we can do about it :/

We should at least have each of these on a newline though